pkg/ee/resource-quota/provider.go

```diff
@@ -77,14 +79,33 @@ func (p *ResourceQuotaProvider) Get(ctx context.Context, userInfo *provider.User
 		return nil, err
 	}
 
-	subject := kubermaticv1.Subject{Name: name, Kind: kind}
-	resourceQuota := &kubermaticv1.ResourceQuota{}
-	if err := masterImpersonatedClient.Get(ctx, types.NamespacedName{
-		Name: buildNameFromSubject(subject),
-	}, resourceQuota); err != nil {
+	subjectNameReq, err := labels.NewRequirement(kubermaticv1.ResourceQuotaSubjectNameLabelKey, selection.Equals, []string{name})
+	if err != nil {
+		return nil, fmt.Errorf("error creating resource quota subject name requirement: %w", err)
+	}
+	subjectKindReq, err := labels.NewRequirement(kubermaticv1.ResourceQuotaSubjectKindLabelKey, selection.Equals, []string{kind})
+	if err != nil {
+		return nil, fmt.Errorf("error creating resource quota subject kind requirement: %w", err)
+	}
+	subjectSelector := labels.NewSelector().Add(*subjectNameReq, *subjectKindReq)
+
+	resourceQuotaList := &kubermaticv1.ResourceQuotaList{}
+	if err := p.privilegedClient.List(ctx,
```

Comment:
Nvmd, I think I get it. It's to support more resources than just projects, right?

Reply:
First we list to find the resource quota for this project. As we dont know the name, we need to list with the filter on subjectName and subjectKind.
For this we need to use a privileged client, as the users wont be able to list (they only have `get` permissions on resource quotas from their projects)

Second request is the GET using the impersonation client so that we make sure that the user can actually access this ResourceQuota.

Reply:
Is there any particular reason why you don't want to relay on resource quota names which are build with a convention of `<kind>-<name>`?

Reply:
Because that convention only works when resource quotas are created through the KKP API.